src/tallies.cpp

```diff
@@ -302,25 +346,89 @@ void Tallies::update_avg_and_var(double x, double& x_avg, double& x_var) {
             ((x_var_old) / (dgen - 1.));
 }
 
+void make_tallies_filter(Tallies& tallies, const YAML::Node& node) {
+  if (!node["tally-filters"]) {
+    fatal_error("tally-filter are not given to evaluate the tallies.");
+  }
+
+  // position-filters
+  if (node["tally-filters"]["position-filters"]) {
+    const YAML::Node& position_filter_nodes =
+        node["tally-filters"]["position-filters"];
+    for (std::size_t i = 0; i < position_filter_nodes.size(); i++) {
+      if (!position_filter_nodes[i]["id"] ||
+          !position_filter_nodes[i]["id"].IsScalar()) {
+        fatal_error(
+            "Invalid \"id\" is given for position filter in tally-filters, the "
+            "possible error entry is " +
+            std::to_string(i) + " from top.");
+      }
+      std::size_t id = position_filter_nodes[i]["id"].as<std::size_t>();
+
+      if (!position_filter_nodes[i]["position-filter-type"] ||
+          !position_filter_nodes[i]["position-filter-type"].IsScalar()) {
```

Comment:
```suggestion
      if (!position_filter_nodes[i]["type"] ||
          !position_filter_nodes[i]["type"].IsScalar()) {
```
The key should just be `type` to simplify input.